airflow/operators/branch.py

```diff
@@ -49,4 +49,6 @@ def choose_branch(self, context: Dict) -> Union[str, Iterable[str]]:
         raise NotImplementedError
 
     def execute(self, context: Dict):
-        self.skip_all_except(context['ti'], self.choose_branch(context))
+        branches_to_execute = self.choose_branch(context)
+        self.skip_all_except(context['ti'], branches_to_execute)
+        return branches_to_execute
```

Comment:
This means it will always push right, even if do_xcom_push is False

Reply:
Not sure about this. I did run the test case that I added for checking xcom push ([Permalink to the test function](https://github.com/apache/airflow/pull/13763/files#diff-fadab52f3011a0eda8eb036bc5df54a5c31396bfc84e467ea0ccf7e01c16842eR173-R193)) with `do_xcom_push=False` and the value returned by -  `ti.xcom_pull(task_ids='make_choice')` was `None`

Have I understood the issue wrong? I followed PythonBranchOperator as a guide to figure out what was needed to be done.

Reply:
Oh yes, 
https://github.com/apache/airflow/blob/3fd5ef355556cf0ad7896bb570bbe4b2eabbf46e/airflow/models/taskinstance.py#L1315-L1318 

should take care of it